src/covid19model/visualization/optimization.py

```diff
@@ -0,0 +1,146 @@
+import datetime
+import random
+import matplotlib
+import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
+import pandas as pd
+import numpy as np
+
+
+# From Color Universal Design (CUD): https://jfly.uni-koeln.de/color/
+colorscale_okabe_ito = {"orange" : "#E69F00", "light_blue" : "#56B4E9",
+                        "green" : "#009E73", "yellow" : "#F0E442",
+                        "blue" : "#0072B2", "red" : "#D55E00",
+                        "pink" : "#CC79A7", "black" : "#000000"}
+
+# covid 19 specific parameters
+plt.rcParams.update({
+    "axes.prop_cycle": plt.cycler('color',
+                                  list(colorscale_okabe_ito.values())),
+    "font.size": 15,
+    "lines.linewidth" : 3,
+    "axes.spines.top": False,
+    "axes.spines.right": False,
+    "ytick.major.left": True,
+    "axes.grid": True
+})
+
+
+def _apply_tick_locator(ax):
+    """support function to apply default ticklocator settings"""
+    ax.xaxis.set_major_locator(plt.MaxNLocator(5))
+    ax.yaxis.set_major_locator(plt.MaxNLocator(4))
+    return ax
+
+def plot_fit(model,data,startDate,states,checkpoints=None,samples=None,filename=None,dataMkr=['o','v','s','*','^'],
+            modelClr=['green','orange','red','black','blue'],legendText=None,titleText=None,getfig=False,ax=None):
+
+    """Plot model fit to user provided data 
+
+    Parameters
+    -----------
+    model: model object
+        correctly initialised model to be fitted to the dataset
+    data: array
+        list containing dataseries
+    startDate: string, format DD-MM-YYY
+        date corresponding to first entry of dataseries
+    states: array
+        list containg the names of the model states that correspond to the data
+    checkpoints : dict, optional
+        A dictionary with a "time" key and additional parameter keys,in the form of
+        ``{"time": [t1, t2, ..], "param": [param1, param2, ..], ..}``
+        indicating new parameter values at the corresponding timestamps.
+    samples: dict, optional
+        A dictionary containing parameter values obtained from a sampling algorithm (f.i. MCMC)
+    filename: string, optional
+        Filename + extension to save a copy of the plot_fit
+    ax : matplotlib.axes.Axes, optional
+        If provided, will use the axis to add the lines.
+
+    Returns
+    -----------
+
+    Notes
+    -----------
+
+    Example use
+    -----------
+
+
+    """
+
+    # Initialize figure and visualize data
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # check if ax object is provided by user
+    if ax is None:
+        fig, ax = plt.subplots()
+    # Create shifted index vector using self.extraTime
+    idx = pd.date_range(startDate,freq='D',periods=data[0].size + model.extraTime) - datetime.timedelta(days=model.extraTime)
```

Comment:
To keep the modules internally more independent and the functionality general, I would leave out the extra_time class attribute and usage of it in this function. As such, the function can be used more general.